torch/_functorch/config.py

```
@@ -140,6 +140,14 @@ def remote_autograd_cache_default() -> Optional[bool]:
 # Generally a good idea since views are free to recompute.
 recompute_views = False
 
+# Rematerialize AC nodes for graphs with forward+loss+backward in one graph.
+# This optimization minimizes activation checkpoint node lifetimes by computing them
+# just-in-time. For AC nodes only used in backward, they are deferred to backward region
+# instead of being computed and saved in forward. This reduces peak memory usage.
+# Note: This only applies to forward+loss+backward graphs where torch.autograd.grad is allowed
+# in the graph. Joint graphs (standard AOTAutograd) use the partitioner instead.
+remat_using_tags_for_fwd_loss_bwd_graph = False
```

One question (maybe for @soulitzer too) - what do we think of either:

(1) making this on by default:

(2) detecting when we've captured a fw-bw-inference graph (by detecting if we are in the inference codepath of AOTAutograd, and any nodes have "backward" metadata, and warning / raising an error if we also detect any AC regions in the graph, that tells users to flip this config?

I'm asking because it seems to me like today, we the current behavior if you capture a fw-bw inference graph containing AC is that we will silently ignore the AC. In which case, this config seems strictly better than the current state (unless we are worried that this config is not tested enough and could introduce silent correctness problems?)

The error is BC breaking, so my tentative vote would be to either turn this on by default, or at least add a warning when we detect the case above and see that the config is not on.